### PR TITLE
GeoIP Region database support

### PR DESCRIPTION
**Pull request by mbyczkowski**
Hi,
I have added support to the GeoIP Region database and it works for me without breaking anything else.
Could you review it and push the gem if you are happy with the change.

Thanks,
Mat

### CODE REVIEW

**Comment by mtodd** [comment]
Sorry for the delay! I'll make sure to look at this tonight/tomorrow and take appropriate action :)

**Comment by mbyczkowski** [comment]
OK. Sounds good. I am also planning to provide lookup by host and ipv6 when
I find some spare time

**Comment by mbyczkowski** [comment]
Hi! Did you have a chance to look at my changes?

File: geoip.c [hunk 1]
```diff
@@ -230,6 +231,57 @@ VALUE rb_geoip_org_look_up(VALUE self, VALUE addr) {
...
+  Check_Type(addr, T_STRING);
+  Data_Get_Struct(self, GeoIP, gi);
+  r = GeoIP_region_by_addr(gi, StringValuePtr(addr));
+  if(r == NULL) return Qnil;
+
+  reg = rb_str_new2(r->region);
+  country = rb_str_new2(r->country_code);
+  
+  country_id = GeoIP_id_by_code(r->country_code);
+  printf("The country id is: %d, for country_code %s\n",country_id, r->country_code);
```
Comment (mtodd) [inline geoip.c]: Probably don't need this.

File: geoip.c [hunk 2]
```diff
@@ -230,6 +231,57 @@ VALUE rb_geoip_org_look_up(VALUE self, VALUE addr) {
...
+  
+  hash = rb_hash_new();
+  rb_hash_sset(hash,"country_code", country);
+  rb_hash_sset(hash,"country_code3", rb_str_new2(GeoIP_code3_by_id(country_id)));
+  rb_hash_sset(hash,"country_name", rb_str_new2(GeoIP_name_by_id(country_id)));
+  rb_hash_sset(hash,"region", reg);
```
Comment (mtodd) [inline geoip.c]: This should look more like:
https://github.com/mtodd/geoip/blob/master/geoip.c#L103-129

Specifically, strings should be wrapped in `encode_to_utf8_and_return_rb_str` to handle encoding correctly. Don't necessarily need to make it a separate function, but that wouldn't hurt.

**Comment by mtodd** [comment]
Thanks for the pull request!

Need some tests for this. I don't have the Regions data file to write some tests for myself...

Other than the comments above, tests are the only thing that will keep me from pulling these in.

**Comment by mbyczkowski** [comment]
Sure, I ll try to make the changes on my branch, and will try to add a few tests as well.

**Comment by alindeman** [comment]
Sorry for not pulling this in yet. Are you interested in rebasing it on current master so it can be merged? Keep the changes to simply the features and tests: I can handle the versioning, etc..

**Comment by mbyczkowski** [comment]
Sure, I will update my branch when I get some spare time.

**Comment by mbyczkowski** [comment]
Closing, as I rebased my changes and I created a new pull request #5

**Pull request closed without merging**